libs/protocol/src/container.rs

```diff
@@ -51,10 +53,27 @@ const FAILS_LIMIT: usize = 10;
 
 const STATS_LIMIT: usize = 30;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LogRecord {
+    pub datetime: NaiveDateTime,
+    pub level: LogLevel,
+    pub message: String,
+}
+
+#[derive(Debug, Clone, AsRefStr, Serialize_repr, Deserialize_repr)]
+#[repr(u16)]
+pub enum LogLevel {
+    Trace = 0,
```

Comment:
did you mean to use a binary bitmap here?

Reply:
I only wanted to use numbers instead of strings, and using separate digits had no particular purpose. However, I like the idea of using a bitmap here. I'll update it.

Reply:
I meant to say bit mask, but I think you got it :)